app/src/main/java/org/mozilla/fenix/browser/OpenInAppOnboardingObserver.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.browser
+
+import android.content.Context
+import android.view.ViewGroup
+import androidx.navigation.NavController
+import mozilla.components.browser.session.Session
+import mozilla.components.feature.app.links.AppLinksUseCases
+import mozilla.components.support.ktx.kotlin.tryGetHostFromUrl
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Displays an [InfoBanner] when a user visits a website that can be opened in an installed native app.
+ *
```

Comment:
Smallest of nits: empty line in this comment block (line that starts with * but is empty)